Lib/pathlib.py

```diff
@@ -562,15 +465,30 @@ def as_uri(self):
         """Return the path as a 'file' URI."""
         if not self.is_absolute():
             raise ValueError("relative path can't be expressed as a file URI")
-        return self._flavour.make_uri(self)
+
+        if self._flavour is posixpath:
+            # On POSIX we represent the path using the local filesystem encoding,
+            # for portability to other applications.
+            return 'file://' + urlquote_from_bytes(bytes(self))
+
+        # Under Windows, file URIs use the UTF-8 encoding.
+        drive = self._drv
+        if len(drive) == 2 and drive[1] == ':':
+            # It's a path on a local drive => 'file:///c:/a/b'
+            rest = self.as_posix()[2:].lstrip('/')
+            return 'file:///%s/%s' % (
+                drive, urlquote_from_bytes(rest.encode('utf-8')))
+        else:
+            # It's a path on a network drive => 'file://host/share/a/b'
+            return 'file:' + urlquote_from_bytes(self.as_posix().encode('utf-8'))
 
     @property
     def _cparts(self):
         # Cached casefolded parts, for hashing and comparison
```

Comment:
There are multiple comments and variable / attribute names using `c` for casefold, they will be harder to understand for new readers.